### PR TITLE
Updating WebJobs SDK references and using updated APIs.

### PR DESCRIPTION
**Pull request by fabiocav**

### CODE REVIEW

File: test/WebJobs.Script.Tests.Integration/EventHubs/EventHubSamplesEndToEndTests.cs [hunk 1]
```diff
@@ -26,7 +26,7 @@ public EventHubSamplesEndToEndTests(TestFixture fixture)
             _fixture = fixture;
         }
 
-        [Fact]
+        [Fact(Skip = "Pending extension installation updates")]
```
Comment (fabiocav) [inline test/WebJobs.Script.Tests.Integration/EventHubs/EventHubSamplesEndToEndTests.cs]: @mathewc - I'm skipping these tests for now. Installation is working, but the process is not currently writing the types as we just make them all internal. Until we change the tooling to use the startup types, we need to keep this disabled.

File: src/WebJobs.Script/Rpc/LanguageWorkerChannel.cs [hunk 2]
```diff
@@ -86,16 +86,17 @@ internal class LanguageWorkerChannel : ILanguageWorkerChannel
...
-            {
-                _eventSubscriptions.Add(_eventManager.OfType<RpcEvent>()
-                    .Where(msg => msg.WorkerId == _workerId)
-                    .Subscribe(msg =>
-                    {
-                        var jsonMsg = JsonConvert.SerializeObject(msg, _verboseSerializerSettings);
-                        _logger.LogTrace(jsonMsg);
-                    }));
-            }
+            // TODO: DI (FACAVAL) BrettSam - This model has been updated, review before merging
```
[the remark this answers is not in the document]
Comment (brettsam) [reply src/WebJobs.Script/Rpc/LanguageWorkerChannel.cs]: I'd just remove the `if` statement. Any filtering will be applied to the logger now via the configured filters that come from the .NET logging infrastructure.

File: test/WebJobs.Script.Tests.Integration/EventHubs/EventHubSamplesEndToEndTests.cs (hunk 1 above)
Comment (pragnagopa) [reply test/WebJobs.Script.Tests.Integration/EventHubs/EventHubSamplesEndToEndTests.cs]: Can we add a TODO or a link to issue to make sure we dont miss enabling these tests?

File: test/WebJobs.Script.Tests.Integration/TestFunctionHost.cs [hunk 3]
```diff
@@ -53,18 +53,19 @@ public TestFunctionHost(string appRoot, Action<IHostBuilder> configureJobHost)
...
                  {
-                     b.ConfigureLogging(l =>
-                     {
-                         l.AddProvider(_loggerProvider);
-                         l.AddFilter<TestLoggerProvider>(_ => true);
-                     })
-                     .AddAzureStorage();
-
-                     configureJobHost?.Invoke(b);
+                     var loggingBuilder = new LoggingBuilder(webJobsBuilder.Services);
```
Comment (brettsam) [inline test/WebJobs.Script.Tests.Integration/TestFunctionHost.cs]: This feels a little clunky. Are you trying to restrict ScriptHostBuilder changes to only happen against the IWebJobsBuilder? Should it expose the IWebHostBuilder as well?

File: src/WebJobs.Script/Rpc/LanguageWorkerChannel.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (fabiocav) [reply src/WebJobs.Script/Rpc/LanguageWorkerChannel.cs]: Great! that's what I was hoping! Will update.

File: test/WebJobs.Script.Tests.Integration/EventHubs/EventHubSamplesEndToEndTests.cs (hunk 1 above)
Comment (fabiocav) [reply test/WebJobs.Script.Tests.Integration/EventHubs/EventHubSamplesEndToEndTests.cs]: Will do. All skipped tests also get logged as skipped and we're going through all of them

File: test/WebJobs.Script.Tests.Integration/TestFunctionHost.cs (hunk 3 above)
Comment (fabiocav) [reply test/WebJobs.Script.Tests.Integration/TestFunctionHost.cs]: The `IWebJobsBuilder` is already exposed (that's what you own and create above). We'd need to pass the `IHostBuilder` instance if we want the extension methods it provides. For most cases, this won't be an issue, but I'll send an update to make this cleaner in a separate PR

File: test/WebJobs.Script.Tests.Integration/WebHostEndToEnd/EndToEndTestsBase.cs [hunk 4]
```diff
@@ -192,7 +192,8 @@ public async Task QueueTriggerToBlobTest()
             string result = await TestHelpers.WaitForBlobAndGetStringAsync(resultBlob);
             Assert.Equal(TestHelpers.RemoveByteOrderMarkAndWhitespace(messageContent), TestHelpers.RemoveByteOrderMarkAndWhitespace(result));
 
-            LogMessage traceEvent = await WaitForTraceAsync(p => p?.FormattedMessage != null && p.FormattedMessage.Contains(id));
+            string userCategory = LogCategories.CreateFunctionUserCategory("QueueTriggerToBlob");
```
Comment (fabiocav) [inline test/WebJobs.Script.Tests.Integration/WebHostEndToEnd/EndToEndTestsBase.cs]: FYI @brettsam , after removing the filter check, I had to add this in order to filter out the worker logs in tests, I believe this is expected, but wanted to bring it up.

**Pull request merged**